src/FakeItEasy/Expressions/ArgumentConstraints/EqualityArgumentConstraint.cs

```diff
@@ -32,7 +36,18 @@ public override string ToString()
                 return "\"{0}\"".FormatInvariant(stringValue);
             }
 
-            return this.ExpectedValue.ToString();
+            try
+            {
+                return this.ExpectedValue.ToString();
+            }
+            catch (Exception)
+            {
+                var type = this.ExpectedValue.GetType();
+
+                return type.Namespace == "Castle.Proxies"
+                    ? $"Faked {type.GetTypeInfo().BaseType.FullName}"
+                    : type.Name;

```

Comment:
For some reason FakeItEasy uses the full name for fakes, whereas non-fakes only use the name:

```C#
Console.WriteLine(new List<int>().ToString());
```
> System.Collections.Generic.List\`1[System.Int32]

```C#
Console.WriteLine(A.Fake<List<int>>().ToString());
```

> Faked System.Collections.Generic.List\`1[[System.Int32, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]]